### ISSUE TITLE
Support building apps using an "Apple Developer" certificate

### ISSUE THREAD

**Issue by trondkr** [microsoft/appcenter#1747]
In my app I allow for "Sign in with Apple" as an Oauth2 sign-in solution for my users. Enabling "Signin with Apple" is now also a requirement from Apple if you have any Oauth2 options enabled. However, it seems only certificates signed with Apple Developer certificates and associated provisioning profile is accepted to use this entitlement (certificates for distribution to App Store). If I try to use Developer ID application "Signin with Apple" is not allowed (https://help.apple.com/developer-account/#/devadf555df9). So the only solution that currently works with 'Signin with Apple' is not supported by Appcenter. It would be great to have this added as a feature request. I think lots of people will be using 'Signin with Apple', but currently its not working with Appcenter. Thanks. Cheers, Trond
ps. If there is any known work-arounds for this I would be happy to hear about them.

**Comment by Oddj0b** [comment]
Hey @trondkr, are you looking for distribute option for your MacOS app that support Signin with Apple?

**Comment by trondkr** [comment]
@Oddj0b Yes correct. If the only option is to distribute through AppStore (for Macos) that is fine but I was hoping I could do that through the Appcenter after initial upload through AppStore.

**Comment by Oddj0b** [comment]
Then I believe it's a duplicate request: https://github.com/microsoft/appcenter/issues/1604

**Comment by Oddj0b** [comment]
if I'm wrong feel free to open the issue again.

**Comment by trondkr** [comment]
@lumaxis To use the entitlement "Signin with Apple" the app has to be code-signed for the App Store which is currently not supported by the appcenter. However, any app that allows the use of signin of app users using OAuth2 functionality are required to include the option "Signin with Apple". So, I can not correctly build and distribute my app using appcenter if I include this entitlement as it requires App Store code signing. I can currently only distribute my app using developer ID code signin on appcenter which will not compile with the entitlement "Signin with Apple"

**Comment by trondkr** [comment]
Yes basically that is correct. T


På 14. april 2020 kl. 10:00:55, Lukas Spieß (notifications@github.com)
skrev:

So, what you’re saying is that you cannot use an Apple Developer
certificate to build your app on App Center?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/microsoft/appcenter/issues/1747#issuecomment-613561818>,
or unsubscribe
